x/stakeibc/keeper/icacallbacks_delegate.go

```diff
@@ -31,12 +31,17 @@ func (k Keeper) UnmarshalDelegateCallbackArgs(ctx sdk.Context, delegateCallback
 	return &unmarshalledDelegateCallback, nil
 }
 
-func DelegateCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, ack *channeltypes.Acknowledgement_Result, args []byte) error {
+func DelegateCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, txMsgData *sdk.TxMsgData, args []byte) error {
 	k.Logger(ctx).Info("DelegateCallback executing", "packet", packet)
 
-	if ack == nil {
-		// transaction on the host chain failed
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "ack is nil, packet %v", packet)
+	if txMsgData == nil {
+		// timeout
+		k.Logger(ctx).Info(fmt.Sprintf("DelegateCallback timeout, ack is nil, packet %v", packet))
+		return nil
+	} else if len(txMsgData.Data) == 0 {
+		// failed transaction
+		k.Logger(ctx).Info(fmt.Sprintf("DelegateCallback tx failed, ack is nil, packet %v", packet))
```

Comment:
sorry one more nit thing: rephrase to "ack is error" for the second case, nil is a little misleading